### DISCUSSION TITLE
How to do backend integration to other server with vite and vike?

### DISCUSSION THREAD

**Discussion by lencil** [vikejs/vike#1493] [Discussion]
In vite official doc, you can integrate any backend server with vite render system.
https://vitejs.dev/guide/backend-integration.html

but, when I use ssr with vike, how to do this? What's the entry of /path/to/main.js in vike?

**Comment by brillout** [comment]
Have you read https://vike.dev/renderPage?

**Reply by lencil** [reply]
Thanks for your reply. I've read this. But it seems that renderPage can only be used with js server. As https://vitejs.dev/guide/backend-integration.html mentioned, It can be used in any server with any programming language. I need to integrate it with rust, c++ or golang backend server. In production it's very easy to do so for the static ssg pages. But in dev environment, now I'm using reverse-proxy to handle the vike dev server. Is there any other way to do so?

**Reply by brillout** [reply]
In dev you always need a Node.js(-like) environment because Vite depends on it. So I'm not sure what you're hoping to achieve?
